### PR TITLE
DAOS-9883 libfabric: Update libfabric with critical performance patches

### PR DESCRIPTION
**Pull request by dmiter**
Update libfabric with critical performance patches.

### CODE REVIEW

**Review by wangshilong — DISMISSED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
You also need to update https://github.com/daos-stack/daos/blob/a7f931a3bbf47561b4faeba4dd70b7403a204851/debian/control#L8

**Comment by wiliamhuang** [comment]
> You also need to update
> 
> https://github.com/daos-stack/daos/blob/a7f931a3bbf47561b4faeba4dd70b7403a204851/debian/control#L8

@brianjmurrell I updated it with "libfabric-dev (>= 1.14.0-2),". I also fixed two conflicting files. 
@daos-stack/daos-gatekeeper Now this PR should be ready to land. All tests passed previously.

**Review by wangshilong — APPROVED** [review]

File: utils/rpms/daos.spec
Comment (dmiter) [inline utils/rpms/daos.spec:557]: The version 2.3.100-2 is already landed. So, you should bump the version everywhere appropriately with next number.

Comment (wiliamhuang) [reply utils/rpms/daos.spec:557]: ok. I am going to update it. Thank you!

**Comment by wiliamhuang** [comment]
@wangshilong @frostedcmos Just made a minor change in the change log in utils/rpms/daos.spec. Could you please approve? Thank you!

File: utils/rpms/daos.spec
Comment (brianjmurrell) [inline utils/rpms/daos.spec:557]: You need to update https://github.com/daos-stack/daos/blob/c24625a9d5a07b790d8335a8423d787f13717fb2/utils/rpms/daos.spec#L30 from `2` to `3` also.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Would probably be worthwhile updating the dates in the `%changelog` and `debian/changelog` to reflect the current date since you have to push again anyway.  Also beware that you are racing with #8608 and are going to conflict with it (and and of course master) if it lands before this PR does.  Perhaps it's worth reaching out to @liw and coordinating your update of this PR once his lands if he anticipates that happening imminently rather than just fixing and pushing again, only to lose the race and have to merge and push again.  It does have 2 +1s on it already so it could land soon.

File: utils/rpms/daos.spec
Comment (wiliamhuang) [reply utils/rpms/daos.spec]: Due to Wei Li's PR just landed, I bumped release to 4.

**Comment by wiliamhuang** [comment]
@brianjmurrell Hi Brian, Shilong and Alex previously approved this PR. I just merged master, updated release number, date of 2.3.100-4 and changelog. Alex is out of office until April 18. Could you please review this PR? I will get Shilong review it too soon. Thank you!

File: utils/rpms/daos.spec
Comment (liw) [reply utils/rpms/daos.spec]: Sorry for the merge conflict. #8608 was fixing a serious correctness issue and the head of a long dependency queue.

**Review by wangshilong — APPROVED** [review]

File: utils/rpms/daos.spec
Comment (wiliamhuang) [reply utils/rpms/daos.spec]: No problem at all. :)

**Review by wiliamhuang — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Comment by wiliamhuang** [comment]
@daos-stack/daos-gatekeeper Please hold off landing this PR at this moment. This PR reverts commit 7ec82992a7c9a4c989e55d8d9a23aa43219e80f3 in libfabric. We will create another PR soon to test Sean Hefty's patch which fixes the issue in 7ec82992a7c9a4c989e55d8d9a23aa43219e80f3. I will update later.

**Comment by brianjmurrell** [comment]
> @daos-stack/daos-gatekeeper Please hold off landing this PR at this moment.

As an additional measure to preventing this PR from landing accidentally because a gatekeeper doesn't see your comment, you could remove the @daos-stack/daos-gatekeeper from the reviewer list.  The best would be if you could -1 your own PR, but GitHub doesn't allow that, unfortunately.

**Review by frostedcmos — APPROVED** [review]

**Comment by wiliamhuang** [comment]
> > @daos-stack/daos-gatekeeper Please hold off landing this PR at this moment.
> 
> As an additional measure to preventing this PR from landing accidentally because a gatekeeper doesn't see your comment, you could remove the @daos-stack/daos-gatekeeper from the reviewer list. The best would be if you could -1 your own PR, but GitHub doesn't allow that, unfortunately.

Thank you, @brianjmurrell! Good to know.

**Pull request closed without merging**
